### PR TITLE
Rework ability effects

### PR DESCRIPTION
**Pull request by jeremylarner**

### CODE REVIEW

File: server/game/Card.js [hunk 1]
```diff
@@ -137,14 +84,52 @@ class Card extends EffectSource {
...
+    }
+
+    get persistentEffects() {
+        let gainedPersistentEffects = this.getEffects('gainAbility').filter(ability => ability.abilityType === 'persistentEffect');
+        if(this.anyEffect('copyCard')) {
+            let mostRecentEffect = _.last(this.effects.filter(effect => effect.type === 'copyCard'));
+            return gainedPersistentEffects.concat(mostRecentEffect.value.getPersistentEffects());
+        }
+
+        return this.isBlank() ? gainedPersistentEffects : gainedPersistentEffects.concat(this.abilities.persistentEffects);
```
Comment (cryogen) [inline server/game/Card.js]: How does this react with the ruling that upgrades that say 'gains: blah blah' blanks the gained stuff too?

File: server/game/Effects/EffectValue.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const _ = require('underscore');
+
+class EffectValue {
+    constructor(value) {
+        this.value = value;
+        if(_.isUndefined(value)) {
```
Comment (cryogen) [inline server/game/Effects/EffectValue.js]: value === undefined is preferable, no need for underscore here

File: server/game/Card.js [hunk 3]
```diff
@@ -356,7 +386,8 @@ class Card extends EffectSource {
         if(this.anyEffect('changeHouse')) {
             combinedHouses = combinedHouses.concat((this.getEffects('changeHouse')));
         } else {
-            combinedHouses.push(this.printedHouse);
+            let copyEffect = this.mostRecentEffect('copyCard');
```
Comment (romanogit) [inline server/game/Card.js]: @jeremylarner : curious, will this cover the fact that Mimic Gel copies a card, but not the house?

Comment (jeremylarner) [reply server/game/Card.js]: No, that will need to be dealt with separately

**Pull request merged**